.github/phpstan.neon

```diff
@@ -30,25 +36,46 @@ parameters:
         - */app/code/core/Mage/Dataflow/Model/Session/Adapter/Http.php
         - */app/code/core/Mage/Dataflow/Model/Convert/Iterator/Http.php
         - */app/code/core/Mage/Dataflow/Model/Convert/Iterator/File/Csv.php
+        - */lib/Varien/Convert/Validator/Column.php
 
         # Cannot declare class Varien_Date, because the name is already in use
         - */app/code/core/Mage/Catalog/Model/Product/Option/Type/Date.php
 
+        # Cannot declare class XMLParser, because the name is already in use
+        - */app/code/core/Mage/Centinel/Model/Api.php
+        - */app/code/core/Mage/Centinel/Model/Api/Client.php
+
         # Class phpseclib\Net\SSH2 not found.
-        - */Paypal/Model/Report/Settlement.php
+        - */app/code/core/Mage/Paypal/Model/Report/Settlement.php
+        - */lib/Varien/Io/Sftp.php
 
         # Class backup/db was not found while trying to analyse it
-        - */Adminhtml/controllers/System/BackupController.php
+        - */app/code/core/Mage/Adminhtml/controllers/System/BackupController.php
 
         # Dataflow/Model/Convert/Iterator.php
-        - */Dataflow/Model/Convert/Iterator.php
+        - */app/code/core/Mage/Dataflow/Model/Convert/Iterator.php
 
         # Too much noise from "Variable $this might not be defined"
         - */app/code/core/*/*/data/*
         - */app/code/core/*/*/sql/*
-    level: 3
+
+        # requires Mycrypt
+        - */lib/Magento/Crypt.php
+        - */lib/Varien/Crypt/Mcrypt.php
+
+        # requires Eaccelerator
+        - */lib/Varien/Cache/Backend/Eaccelerator.php
+
+        # Instantiated class not found
+        - */lib/Varien/Image/Adapter.php
+
+        # todo check
+        - */app/code/core/Mage/Eav/Model/Convert/Adapter/Grid.php
+        - */app/code/core/Mage/Weee/Model/Total/Quote/Weee.php
+    level: 5
     checkFunctionNameCase: true
     checkInternalClassCaseSensitivity: true
+    treatPhpDocTypesAsCertain: false
```

Comment:
what does it do in practice?

Reply:
Finally it ignores some errors. One line i remember is ...

```
        /** @var Mage_Page_Block_Html_Head $head */
        $head = $this->getLayout()->getBlock('head');
        if ($head) {
```
Related error:

```
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Cms/Block/Page.php                                                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  99     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  105    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
```

... caused by `getBlock('head');` that returns `false|Mage_Core_Block_Abstract`

I could add `false` as return type too, but we have a couple of other errors that could/should be fixed first. `treatPhpDocTypesAsCertain` can be removed later. 


### All changes to baseline


```
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Admin/Model/Config.php                                                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  48     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/Block/Customer/Edit/Tab/View/Cart.php                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  60     Ternary operator condition is always true.                                                                                                          
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/Block/Notification/Toolbar.php                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  156    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/Block/Report/Filter/Form.php                                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  209    Result of && is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  220    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/Block/System/Email/Template/Edit/Form.php                                                                                                 
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  41     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/Block/Widget/Form.php                                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  173    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------- 
  Line   Adminhtml/Block/Widget/Grid.php                                        
 ------ ----------------------------------------------------------------------- 
  350    Else branch is unreachable because previous condition is always true.  
 ------ ----------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/Block/Widget/Grid/Massaction/Item.php                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  66     Call to function is_array() with Mage_Core_Block_Abstract will always evaluate to false.                                                            
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------- 
  Line   Adminhtml/Model/Sales/Order/Create.php                                 
 ------ ----------------------------------------------------------------------- 
  1288   Else branch is unreachable because previous condition is always true.  
 ------ ----------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/Model/System/Store.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  381    Call to function is_array() with int will always evaluate to false.                                                                                 
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  468    Call to function is_array() with int will always evaluate to false.                                                                                 
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/controllers/Catalog/Product/AttributeController.php                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  290    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/controllers/Catalog/ProductController.php                                                                                                 
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  703    Call to function is_null() with array will always evaluate to false.                                                                                
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Adminhtml/controllers/IndexController.php                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  376    Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  387    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------- 
  Line   Api/Model/Server/Handler/Abstract.php                                          
 ------ ------------------------------------------------------------------------------- 
  288    Else branch is unreachable because ternary operator condition is always true.  
 ------ ------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Api/Model/Server/Wsi/Handler.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  91     Call to function is_object() with string will always evaluate to false.                                                                             
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Api2/Model/Acl/Filter/Attribute/ResourcePermission.php                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  109    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Api2/Model/Acl/Global/Rule/ResourcePermission.php                                                                                                   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  107    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Api2/Model/Renderer/Xml.php                                                                                                                         
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  95     Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Authorizenet/Model/Directpost/Observer.php                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  57     Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Bundle/Block/Catalog/Product/View/Type/Bundle/Option.php                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  213    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Bundle/Block/Sales/Order/Items/Renderer.php                                                                                                         
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  37     If condition is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  80     If condition is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Bundle/Model/Observer.php                                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  130    Elseif branch is unreachable because previous condition is always true.                                                                             
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  130    Instanceof between *NEVER* and Varien_Object will always evaluate to false.                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Bundle/Model/Option.php                                                                                                                                                                     
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  73     Ignored error pattern #^Negated boolean expression is always false\.$# in path /var/www/html/app/code/core/Mage/Bundle/Model/Option.php is expected to occur 1 time, but occurred 2 times.  
  111    Negated boolean expression is always false.                                                                                                                                                 
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Bundle/Model/Product/Price.php                                                                                                                                                                                       
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  125    Ignored error pattern #^Call to function is_null\(\) with float will always evaluate to false\.$# in path /var/www/html/app/code/core/Mage/Bundle/Model/Product/Price.php is expected to occur 1 time, but occurred  
         4 times.                                                                                                                                                                                                             
  571    Call to function is_null() with float will always evaluate to false.                                                                                                                                                 
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                                    
  690    Call to function is_null() with float will always evaluate to false.                                                                                                                                                 
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                                    
  972    Call to function is_null() with float will always evaluate to false.                                                                                                                                                 
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                                    
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Bundle/Model/Product/Type.php                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  397    Else branch is unreachable because ternary operator condition is always true.                                                                       
  949    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Block/Breadcrumbs.php                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  52     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Block/Category/View.php                                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  43     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Block/Navigation.php                                                                                                                        
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  183    Else branch is unreachable because previous condition is always true.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Block/Product/Abstract.php                                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  672    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Block/Product/List/Toolbar.php                                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  474    Call to function is_null() with string will always evaluate to false.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Block/Product/View.php                                                                                                                      
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  56     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Block/Seo/Sitemap/Abstract.php                                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  40     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Block/Seo/Sitemap/Tree/Category.php                                                                                                         
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  69     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------- 
  Line   Catalog/Helper/Category.php                            
 ------ ------------------------------------------------------- 
  98     Unreachable statement - code above always terminates.  
 ------ ------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Helper/Output.php                                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  126    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  162    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Helper/Product/View.php                                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  84     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Category/Indexer/Flat.php                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  124    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  138    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------- 
  Line   Catalog/Model/Convert/Adapter/Product.php           
 ------ ---------------------------------------------------- 
  360    Offset string on string in isset() does not exist.  
 ------ ---------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Design.php                                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  347    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  371    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Product/Api/V2.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  106    Call to function is_object() with array will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  106    Result of || is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  245    Call to function is_object() with array will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Product/Attribute/Api.php                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  101    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Product/Image.php                                                                                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  376    Ignored error pattern #^Negated boolean expression is always false\.$# in path /var/www/html/app/code/core/Mage/Catalog/Model/Product/Image.php is expected to occur 1 time, but occurred 2 times.  
  452    Negated boolean expression is always false.                                                                                                                                                         
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Product/Indexer/Flat.php                                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  132    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  138    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  144    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  146    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  161    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  170    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Product/Type/Abstract.php                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  886    Ternary operator condition is always true.                                                                                                          
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Product/Type/Configurable.php                                                                                                                                                       
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  714    Ignored error pattern #^If condition is always true\.$# in path /var/www/html/app/code/core/Mage/Catalog/Model/Product/Type/Configurable.php is expected to occur 2 times, but occurred 4 times.  
  829    If condition is always true.                                                                                                                                                                      
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                 
  832    If condition is always true.                                                                                                                                                                      
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  Line   Catalog/Model/Product/Type/Price.php                                                                                                                                                                              
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  73     Ignored error pattern #^Call to function is_null\(\) with float will always evaluate to false\.$# in path /var/www/html/app/code/core/Mage/Catalog/Model/Product/Type/Price.php is expected to occur 1 time, but  
         occurred 2 times.                                                                                                                                                                                                 
  434    Call to function is_null() with float will always evaluate to false.                                                                                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                                 
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 

 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Resource/Category/Collection.php                                                                                                                                                     
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
         Ignored error pattern #^Variable \$condition might not be defined\.$# in path /var/www/html/app/code/core/Mage/Catalog/Model/Resource/Category/Collection.php was not matched in reported errors.  
  125    Elseif branch is unreachable because previous condition is always true.                                                                                                                            
  127    Elseif branch is unreachable because previous condition is always true.                                                                                                                            
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Resource/Category/Flat.php                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  181    Call to function is_string() with int will always evaluate to false.                                                                                
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------- 
  Line   Catalog/Model/Resource/Category/Flat/Collection.php                    
 ------ ----------------------------------------------------------------------- 
  176    Else branch is unreachable because previous condition is always true.  
 ------ ----------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Resource/Category/Tree.php                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  378    Call to function is_string() with true will always evaluate to false.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Resource/Product.php                                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  392    Call to function is_array() with Mage_Catalog_Model_Product will always evaluate to false.                                                          
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  396    Call to function is_array() with null will always evaluate to false.                                                                                
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  396    Result of && is always false.                                                                                                                       
  423    Call to function is_numeric() with Mage_Catalog_Model_Product will always evaluate to false.                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  424    Call to function is_numeric() with Mage_Core_Model_Store will always evaluate to false.                                                             
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  486    Call to function is_array() with Mage_Catalog_Model_Product|null will always evaluate to false.                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  486    Result of && is always false.                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Resource/Product/Collection.php                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  412    Instanceof between Mage_Eav_Model_Entity_Abstract and Mage_Core_Model_Resource_Db_Abstract will always evaluate to false.                           
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  412    Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  890    Else branch is unreachable because ternary operator condition is always true.                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Resource/Product/Type/Configurable.php                                                                                                
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  53     Else branch is unreachable because previous condition is always true.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  63     Right side of && is always true.                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Resource/Product/Website.php                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  56     Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  90     Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/Model/Resource/Url.php                                                                                                                      
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  302    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------ 
  Line   Catalog/Model/Url.php                                                               
 ------ ------------------------------------------------------------------------------------ 
  720    Strict comparison using === between string and null will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------ 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Catalog/controllers/CategoryController.php                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  166    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogIndex/Model/Resource/Indexer.php                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  92     Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogInventory/Model/Indexer/Stock.php                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  128    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  136    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  209    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogInventory/Model/Observer.php                                                                                                                                                                      
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  307    Ignored error pattern #^Negated boolean expression is always false\.$# in path /var/www/html/app/code/core/Mage/CatalogInventory/Model/Observer.php is expected to occur 2 times, but occurred 3 times.  
  322    Ignored error pattern #^If condition is always true\.$# in path /var/www/html/app/code/core/Mage/CatalogInventory/Model/Observer.php is expected to occur 4 times, but occurred 5 times.                 
  516    Negated boolean expression is always false.                                                                                                                                                              
  806    Ternary operator condition is always true.                                                                                                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                        
  977    If condition is always true.                                                                                                                                                                             
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                        
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogInventory/Model/Resource/Stock/Item/Collection.php                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  64     Else branch is unreachable because previous condition is always true.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------- 
  Line   CatalogInventory/Model/Stock/Status.php  
 ------ ----------------------------------------- 
  116    Result of && is always false.            
 ------ ----------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogSearch/Block/Advanced/Form.php                                                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  39     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogSearch/Block/Advanced/Result.php                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  40     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogSearch/Block/Result.php                                                                                                                      
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  65     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogSearch/Model/Advanced.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  113    Call to function is_null() with Mage_Catalog_Model_Resource_Product_Attribute_Collection will always evaluate to false.                             
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogSearch/Model/Indexer/Fulltext.php                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  145    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  160    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  169    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   CatalogSearch/Model/Resource/Helper/Mysql4.php                                                                                                      
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  89     Elseif condition is always true.                                                                                                                    
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Checkout/Block/Cart/Item/Renderer.php                                                                                                                                                      
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  365    Ignored error pattern #^If condition is always true\.$# in path /var/www/html/app/code/core/Mage/Checkout/Block/Cart/Item/Renderer.php is expected to occur 1 time, but occurred 2 times.  
  368    If condition is always true.                                                                                                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                          
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------- 
  Line   Checkout/Block/Cart/Totals.php                                                 
 ------ ------------------------------------------------------------------------------- 
  93     Else branch is unreachable because ternary operator condition is always true.  
 ------ ------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Checkout/Model/Api/Resource/Product.php                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  123    Call to function is_null() with Mage_Sales_Model_Quote_Item|false will always evaluate to false.                                                    
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Checkout/Model/Cart/Customer/Api/V2.php                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  53     Call to function is_array() with object will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Checkout/Model/Type/Multishipping.php                                                                                                                                                       
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  162    Ignored error pattern #^If condition is always true\.$# in path /var/www/html/app/code/core/Mage/Checkout/Model/Type/Multishipping.php is expected to occur 5 times, but occurred 6 times.  
  446    If condition is always true.                                                                                                                                                                
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Cms/Block/Page.php                                                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  99     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  105    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Cms/Model/Page.php                                                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  105    Call to function is_null() with int|string will always evaluate to false.                                                                           
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   ConfigurableSwatches/Helper/Data.php                                                                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  49     Ignored error pattern #^Right side of && is always true\.$# in path /var/www/html/app/code/core/Mage/ConfigurableSwatches/Helper/Data.php is expected to occur 1 time, but occurred 2 times.  
  130    Right side of && is always true.                                                                                                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Block/Abstract.php                                                                                                                                                                      
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  448    Call to function is_string() with Mage_Core_Block_Abstract will always evaluate to false.                                                                                                    
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                            
  451    Ignored error pattern #^Negated boolean expression is always false\.$# in path /var/www/html/app/code/core/Mage/Core/Block/Abstract.php is expected to occur 2 times, but occurred 3 times.  
  675    Negated boolean expression is always false.                                                                                                                                                  
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Controller/Front/Action.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  139    Call to function is_null() with array|string will always evaluate to false.                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Controller/Front/Router.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  71     Call to function is_string() with array will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Controller/Varien/Action.php                                                                                                                   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  1084   Call to function is_null() with array|string will always evaluate to false.                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Helper/Data.php                                                                                                                                
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  460    Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  460    Result of || is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  461    Result of || is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  529    Result of || is always true.                                                                                                                        
  549    Elseif branch is unreachable because previous condition is always true.                                                                             
  581    Else branch is unreachable because previous condition is always true.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  716    Strict comparison using === between string and null will always evaluate to false.                                                                  
  719    Strict comparison using === between string and true will always evaluate to false.                                                                  
  722    Strict comparison using === between string and false will always evaluate to false.                                                                 
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------- 
  Line   Core/Model/App.php                                                     
 ------ ----------------------------------------------------------------------- 
  1406   Else branch is unreachable because previous condition is always true.  
 ------ ----------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Model/App/Emulation.php                                                                                                                        
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  72     Call to function is_null() with string will always evaluate to false.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------- 
  Line   Core/Model/Cookie.php                                                                 
 ------ -------------------------------------------------------------------------------------- 
  227    Strict comparison using === between int|null and true will always evaluate to false.  
 ------ -------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------- 
  Line   Core/Model/Design/Package.php                       
 ------ ---------------------------------------------------- 
  273    Offset string on string in empty() does not exist.  
 ------ ---------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Model/Email/Template.php                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  349    Right side of && is always false.                                                                                                                   
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Model/Layout/Update.php                                                                                                                        
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  156    Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Model/Locale.php                                                                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  660    Call to function is_null() with float|int|string will always evaluate to false.                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Model/Resource/Helper/Mysql4.php                                                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  309    Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------- 
  Line   Core/Model/Resource/Iterator.php                       
 ------ ------------------------------------------------------- 
  83     Unreachable statement - code above always terminates.  
 ------ ------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------------- 
  Line   Core/Model/Resource/Translate/String.php                                               
 ------ --------------------------------------------------------------------------------------- 
  171    Strict comparison using === between int|null and false will always evaluate to false.  
 ------ --------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Core/Model/Resource/Website/Collection.php                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  103    Call to function is_array() with int will always evaluate to false.                                                                                 
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------ 
  Line   Core/Model/Session/Abstract.php                                                     
 ------ ------------------------------------------------------------------------------------ 
  505    Strict comparison using === between string and true will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------ 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Customer/Helper/Address.php                                                                                                                         
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  184    Ternary operator condition is always true.                                                                                                          
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  193    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Customer/Helper/Data.php                                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  616    Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  616    Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  616    Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------ 
  Line   Customer/Model/Address/Abstract.php                                                                   
 ------ ------------------------------------------------------------------------------------------------------ 
  147    Strict comparison using === between int<min, -2>|int<1, max> and null will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------------------------ 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Customer/Model/Customer/Api.php                                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  87     Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Customer/controllers/AccountController.php                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  619    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  930    Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  941    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Dataflow/Model/Convert/Action/Abstract.php                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  82     Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  82     Result of && is always false.                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Dataflow/Model/Resource/Import.php                                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  60     Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------- 
  Line   Directory/Helper/Url.php                                                       
 ------ ------------------------------------------------------------------------------- 
  41     Else branch is unreachable because ternary operator condition is always true.  
 ------ ------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Directory/Model/Currency.php                                                                                                                        
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  340    Instanceof between string and Mage_Directory_Model_Currency will always evaluate to false.                                                          
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Directory/Model/Resource/Currency/Collection.php                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  99     Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------- 
  Line   Downloadable/Model/Link/Api/Validator.php                                      
 ------ ------------------------------------------------------------------------------- 
  201    Else branch is unreachable because ternary operator condition is always true.  
  211    Else branch is unreachable because ternary operator condition is always true.  
  241    Else branch is unreachable because ternary operator condition is always true.  
  269    Else branch is unreachable because ternary operator condition is always true.  
 ------ ------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------- 
  Line   Eav/Model/Attribute/Data/Abstract.php                                                
 ------ ------------------------------------------------------------------------------------- 
  191    Strict comparison using === between string and false will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Attribute/Data/Date.php                                                          
 ------ ------------------------------------------------------------------------------------------- 
  56     Strict comparison using === between array|string and false will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Attribute/Data/Multiline.php                                                     
 ------ ------------------------------------------------------------------------------------------- 
  60     Strict comparison using === between array|string and false will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Attribute/Data/Select.php                                                        
 ------ ------------------------------------------------------------------------------------------- 
  55     Strict comparison using === between array|string and false will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Attribute/Data/Text.php                                                          
 ------ ------------------------------------------------------------------------------------------- 
  56     Strict comparison using === between array|string and false will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Convert/Adapter/Entity.php                                                                                                                
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  222    Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  224    Offset 'attribute' on *NEVER* on left side of ?? always exists and is always null.                                                                  
  225    Offset 'attribute' on *NEVER* in isset() always exists and is always null.                                                                          
  226    Offset 'bind' on *NEVER* on left side of ?? always exists and is always null.                                                                       
  227    Offset 'joinType' on *NEVER* on left side of ?? always exists and is always null.                                                                   
  228    Offset 'storeId' on *NEVER* on left side of ?? always exists and is always null.                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Entity/Abstract.php                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  384    Instanceof between Mage_Core_Model_Config_Element and Mage_Eav_Model_Entity_Attribute_Abstract will always evaluate to false.                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Entity/Attribute/Backend/Abstract.php                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  176    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  202    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Entity/Collection/Abstract.php                                                                                                                                                         
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
         Ignored error pattern #^Parameter \#2 \$array of function implode expects array<string>, array<int, array<int, string>\|string> given\.$# in path                                                
         /var/www/html/app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php was not matched in reported errors.                                                                                    
         Ignored error pattern #^Variable \$bindAttribute might not be defined\.$# in path /var/www/html/app/code/core/Mage/Eav/Model/Entity/Collection/Abstract.php was not matched in reported errors.  
  202    Elseif branch is unreachable because previous condition is always true.                                                                                                                          
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                
  202    Instanceof between *NEVER* and Mage_Core_Model_Config_Element will always evaluate to false.                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                
  202    Result of || is always true.                                                                                                                                                                     
  204    Else branch is unreachable because previous condition is always true.                                                                                                                            
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                
  297    Strict comparison using === between array|int|Mage_Eav_Model_Entity_Attribute_Interface|string and null will always evaluate to false.                                                           
  503    Call to function is_array() with string will always evaluate to false.                                                                                                                           
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                
  811    Else branch is unreachable because previous condition is always true.                                                                                                                            
  1427   Call to function is_array() with string will always evaluate to false.                                                                                                                           
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Entity/Setup.php                                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  849    Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  897    Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Resource/Entity/Attribute.php                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  492    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Eav/Model/Resource/Entity/Attribute/Collection.php                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  120    Call to function is_array() with int will always evaluate to false.                                                                                 
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   GiftMessage/Helper/Message.php                                                                                                                      
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  169    Call to function is_null() with int will always evaluate to false.                                                                                  
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  169    Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  169    Strict comparison using === between int and '' will always evaluate to false.                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------- 
  Line   GiftMessage/Model/Api.php                              
 ------ ------------------------------------------------------- 
  42     Unreachable statement - code above always terminates.  
 ------ ------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   ImportExport/Block/Adminhtml/Import/Frame/Result.php                                                                                                
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  69     Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  91     Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Index/Model/Process.php                                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  167    Result of && is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  170    Unreachable statement - code above always terminates.                                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Index/Model/Resource/Event/Collection.php                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  91     Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  91     Result of && is always false.                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Index/controllers/Adminhtml/ProcessController.php                                                                                                   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  72     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  97     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  131    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  188    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Log/Model/Customer.php                                                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  62     Instanceof between int|Mage_Log_Model_Customer and Mage_Customer_Model_Customer will always evaluate to false.                                      
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Log/Model/Visitor.php                                                                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  284    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  315    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  332    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  392    Right side of && is always true.                                                                                                                    
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Newsletter/Model/Resource/Template.php                                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  46     Call to function is_null() with string will always evaluate to false.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Oauth/controllers/Adminhtml/Oauth/AuthorizeController.php                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  160    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Page/Block/Html/Head.php                                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  285    Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  579    Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  579    Strict comparison using === between string and true will always evaluate to false.                                                                  
  585    Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  585    Strict comparison using === between string and true will always evaluate to false.                                                                  
  591    Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  591    Strict comparison using === between string and false will always evaluate to false.                                                                 
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Page/Block/Template/Links.php                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  91     Call to function is_null() with string will always evaluate to false.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  91     Result of || is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  91     Strict comparison using === between string and false will always evaluate to false.                                                                 
  96     Else branch is unreachable because ternary operator condition is always true.                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Paygate/Helper/Data.php                                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  99     Call to function is_null() with string will always evaluate to false.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------- 
  Line   Payment/Helper/Data.php                                
 ------ ------------------------------------------------------- 
  95     Unreachable statement - code above always terminates.  
 ------ ------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Paypal/Block/Express/Shortcut.php                                                                                                                   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  97     Call to function is_null() with Mage_Catalog_Model_Product will always evaluate to false.                                                           
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------- 
  Line   Paypal/Controller/Express/Abstract.php                                               
 ------ ------------------------------------------------------------------------------------- 
  480    Strict comparison using === between string and false will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------- 
  Line   Paypal/Model/Direct.php                                                                                
 ------ ------------------------------------------------------------------------------------------------------- 
  87     Strict comparison using === between int|Mage_Core_Model_Store and null will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------- 
  Line   Paypal/Model/Express.php                                                                               
 ------ ------------------------------------------------------------------------------------------------------- 
  121    Strict comparison using === between int|Mage_Core_Model_Store and null will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  Line   Paypal/Model/Express/Checkout.php                                                                                                                                                                     
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  501    Ignored error pattern #^Negated boolean expression is always false\.$# in path /var/www/html/app/code/core/Mage/Paypal/Model/Express/Checkout.php is expected to occur 1 time, but occurred 2 times.  
  614    Negated boolean expression is always false.                                                                                                                                                           
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                     
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 

 ------ ------------------------------------------------------------------------------------------------------- 
  Line   Paypal/Model/Method/Agreement.php                                                                      
 ------ ------------------------------------------------------------------------------------------------------- 
  87     Strict comparison using === between int|Mage_Core_Model_Store and null will always evaluate to false.  
 ------ ------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Paypal/Model/Observer.php                                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  89     Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Persistent/Model/Observer.php                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  71     Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  206    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  229    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  280    Unreachable statement - code above always terminates.                                                                                               
  443    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  545    Right side of && is always true.                                                                                                                    
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Persistent/Model/Observer/Session.php                                                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  41     Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  94     Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  122    Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Poll/Model/Resource/Poll/Answer/Collection.php                                                                                                      
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  60     Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Rating/Model/Observer.php                                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  41     Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Rating/Model/Resource/Rating/Collection.php                                                                                                         
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  97     Call to function is_array() with int will always evaluate to false.                                                                                 
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  98     Strict comparison using === between int and null will always evaluate to false.                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Rating/Model/Resource/Rating/Option/Vote/Collection.php                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  117    Call to function is_array() with int|int<1, max> will always evaluate to false.                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Reports/Model/Event/Observer.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  233    Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Reports/Model/Resource/Product/Lowstock/Collection.php                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  180    Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Review/Model/Observer.php                                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  57     Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Review/Model/Resource/Review.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  402    Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Review/controllers/ProductController.php                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  240    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Rss/Block/List.php                                                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  46     Left side of && is always true.                                                                                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Adminhtml/Report/Filter/Form.php                                                                                                        
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  44     Result of && is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Adminhtml/Report/Filter/Form/Coupon.php                                                                                                 
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  46     Result of && is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Adminhtml/Report/Filter/Form/Order.php                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  42     Result of && is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Order/Creditmemo.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  44     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Order/Info.php                                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  46     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Order/Invoice.php                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  44     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Order/Print.php                                                                                                                         
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  38     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Order/Print/Creditmemo.php                                                                                                              
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  38     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Order/Print/Invoice.php                                                                                                                 
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  38     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Order/Print/Shipment.php                                                                                                                
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  76     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Order/Shipment.php                                                                                                                      
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  44     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Block/Order/View.php                                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  44     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Entity/Order/Creditmemo/Collection.php                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  44     Else branch is unreachable because previous condition is always true.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Entity/Order/Invoice/Collection.php                                                                                                     
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  44     Else branch is unreachable because previous condition is always true.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Entity/Order/Shipment/Collection.php                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  44     Else branch is unreachable because previous condition is always true.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Observer.php                                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  117    Call to function is_numeric() with Mage_Catalog_Model_Product will always evaluate to false.                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  128    Else branch is unreachable because previous condition is always true.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Order/Creditmemo.php                                                                                                                    
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  708    Instanceof between string and Mage_Sales_Model_Order_Creditmemo_Comment will always evaluate to false.                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Order/Item.php                                                                                                                          
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  272    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  666    Elseif branch is unreachable because previous condition is always true.                                                                             
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Order/Payment.php                                                                                                                       
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  450    Call to function is_null() with Mage_Sales_Model_Order_Invoice will always evaluate to false.                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  1433   Else branch is unreachable because ternary operator condition is always true.                                                                       
  1660   Call to function is_array() with string will always evaluate to false.                                                                              
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------- 
  Line   Sales/Model/Order/Payment/Transaction.php              
 ------ ------------------------------------------------------- 
  618    Unreachable statement - code above always terminates.  
 ------ ------------------------------------------------------- 

 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Order/Pdf/Abstract.php                                                                                                                                                    
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
         Ignored error pattern #^Variable \$order might not be defined\.$# in path /var/www/html/app/code/core/Mage/Sales/Model/Order/Pdf/Abstract.php was not matched in reported errors.     
         Ignored error pattern #^Variable \$shipment might not be defined\.$# in path /var/www/html/app/code/core/Mage/Sales/Model/Order/Pdf/Abstract.php was not matched in reported errors.  
  269    Elseif branch is unreachable because previous condition is always true.                                                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                     
  269    Instanceof between *NEVER* and Mage_Sales_Model_Order_Shipment will always evaluate to false.                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                     
 ------ -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Order/Shipment/Api/V2.php                                                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  81     If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Quote/Item.php                                                                                                                                                                     
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  322    Ignored error pattern #^Negated boolean expression is always false\.$# in path /var/www/html/app/code/core/Mage/Sales/Model/Quote/Item.php is expected to occur 1 time, but occurred 2 times.  
  444    Negated boolean expression is always false.                                                                                                                                                    
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                              
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  Line   Sales/Model/Quote/Item/Abstract.php                                                                                                                                                       
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  253    Ignored error pattern #^If condition is always true\.$# in path /var/www/html/app/code/core/Mage/Sales/Model/Quote/Item/Abstract.php is expected to occur 5 times, but occurred 6 times.  
  888    If condition is always true.                                                                                                                                                              
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Sales/Model/Resource/Quote/Item/Option/Collection.php                                                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  96     Instanceof between int<min, -1>|int<1, max> and Mage_Sales_Model_Quote_Item will always evaluate to false.                                          
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   SalesRule/Model/Observer.php                                                                                                                                                                     
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  47     Ignored error pattern #^Negated boolean expression is always false\.$# in path /var/www/html/app/code/core/Mage/SalesRule/Model/Observer.php is expected to occur 1 time, but occurred 2 times.  
  77     Negated boolean expression is always false.                                                                                                                                                      
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.                                                
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------- 
  Line   SalesRule/Model/Rule.php                               
 ------ ------------------------------------------------------- 
  500    Unreachable statement - code above always terminates.  
 ------ ------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Shipping/Helper/Data.php                                                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  88     Call to function is_object() with non-falsy-string will always evaluate to false.                                                                   
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  103    Call to function is_object() with non-falsy-string will always evaluate to false.                                                                   
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  118    Call to function is_object() with non-falsy-string will always evaluate to false.                                                                   
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Shipping/Model/Config.php                                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  111    Negated boolean expression is always false.                                                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  Line   Shipping/Model/Shipping.php                                                                                                                                                                     
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 
  156    Ignored error pattern #^Negated boolean expression is always false\.$# in path /var/www/html/app/code/core/Mage/Shipping/Model/Shipping.php is expected to occur 1 time, but occurred 2 times.  
  443    Negated boolean expression is always false.                                                                                                                                                     
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Tax/Helper/Data.php                                                                                                                                 
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  1004   Right side of && is always true.                                                                                                                    
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  1118   Right side of && is always true.                                                                                                                    
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Tax/Model/Calculation.php                                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  401    Result of && is always false.                                                                                                                       
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  401    Strict comparison using === between int|null and false will always evaluate to false.                                                               
  462    Strict comparison using === between int|null and false will always evaluate to false.                                                               
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------------- 
  Line   Widget/Model/Widget/Config.php                                                 
 ------ ------------------------------------------------------------------------------- 
  126    Else branch is unreachable because ternary operator condition is always true.  
 ------ ------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Wishlist/Model/Item.php                                                                                                                             
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  604    Call to function is_array() with Mage_Wishlist_Model_Item_Option will always evaluate to false.                                                     
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  609    Elseif branch is unreachable because previous condition is always true.                                                                             
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  609    Instanceof between *NEVER* and Varien_Object will always evaluate to false.                                                                         
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
  613    Else branch is unreachable because previous condition is always true.                                                                               
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Wishlist/Model/Resource/Item/Collection.php                                                                                                         
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  207    If condition is always true.                                                                                                                        
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   Wishlist/Model/Resource/Item/Option/Collection.php                                                                                                  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 
  94     Instanceof between int<min, -1>|int<1, max> and Mage_Wishlist_Model_Item will always evaluate to false.                                             
         💡 Because the type is coming from a PHPDoc, you can turn off this check by setting treatPhpDocTypesAsCertain: false in your .github/phpstan.neon.   
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------- 

 [ERROR] Found 316 errors                                                                                               


```